processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -1099,16 +1101,17 @@ private PropertyMapping(String name, String targetWriteAccessorName,
         ReadAccessor targetReadAccessorProvider,
         Type targetType, Assignment propertyAssignment,
         Set<String> dependsOn, Assignment defaultValueAssignment, boolean constructorMapping) {
-        this( name, null, targetWriteAccessorName, targetReadAccessorProvider,
+        this( name, name, null, targetWriteAccessorName, targetReadAccessorProvider,
```

Comment:
I don't think that we need to pass the name as a source here. Constant expressions do not have a source property.